File: java/src/main/java/com/sopra/TPFinal/model/Salle.java
package com.sopra.TPFinal.model;

import java.util.Set;

import javax.persistence.Column;
import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.OneToMany;

import com.fasterxml.jackson.annotation.JsonView;
import com.sopra.TPFinal.model.view.JsonViews;

@Entity
@DiscriminatorValue("salle")
public class Salle extends Materiel {
	@JsonView(JsonViews.Common.class)
	@Column(name = "capacite")
	private Integer capacite;
	@OneToMany(fetch = FetchType.LAZY)
	private Set<Formation> formations;
	
	public Salle() {
		super();
	}

	public Integer getCapacite() {
		return capacite;
	}

	public void setCapacite(Integer capacite) {
		this.capacite = capacite;
	}

}

File: angular/src/app/model/salle.ts
import {Formation} from './Formation';
import {Materiel} from './materiel';

export class Salle extends Materiel {

  constructor(private _capacite?: number, private _formationsSalle?: Formation[]){
    super();
  }


  get formationsSalle(): Formation[] {
    return this._formationsSalle;
  }

  set formationsSalle(value: Formation[]) {
    this._formationsSalle = value;
  }

  get capacite(): number {
    return this._capacite;
  }

  set capacite(value: number) {
    this._capacite = value;
  }
}
